src/Models/Dspace.php

```diff
@@ -0,0 +1,337 @@
+<?php
+
+/**
+ * @author Nicolas CARPi <nico-git@deltablot.email>
+ * @author Moustapha <Deltablot>
+ * @copyright 2025 Nicolas CARPi
+ * @see https://www.elabftw.net Official website
+ * @license AGPL-3.0
+ * @package elabftw
+ */
+declare(strict_types=1);
+
+namespace Elabftw\Models;
+
+use Defuse\Crypto\Crypto;
+use Defuse\Crypto\Key;
+use Elabftw\Elabftw\Env;
+use Elabftw\Enums\Action;
+use Elabftw\Enums\DspaceAction;
+use Elabftw\Enums\EntityType;
+use Elabftw\Enums\Storage;
+use Elabftw\Exceptions\ImproperActionException;
+use Elabftw\Interfaces\QueryParamsInterface;
+use Elabftw\Make\MakeEln;
+use Elabftw\Models\Users\Users;
+use Elabftw\Services\HttpGetter;
+use Override;
+use ZipStream\ZipStream;
+
+use function str_starts_with;
+use function rtrim;
+use function sprintf;
+use function json_decode;
+
+/**
+ * Connect with DSpace Repository
+ * https://dspace.org/
+ */
+final class Dspace extends AbstractRest
+{
+    private const METADATA_SECTIONS = array(
+        'dc.contributor.author' => 'publicationStep',
+        'dc.title' => 'publicationStep',
+        'dc.date.issued' => 'publicationStep',
+        'dc.type' => 'publicationStep',
+        'dc.description.abstract' => 'traditionalpagetwo',
+    );
+
+    private ?array $headers = null;
+
+    private string $host;
+
+    // rename host to host
+    public function __construct(
+        private readonly Users $requester,
+        private readonly HttpGetter $httpGetter,
+        string $host,
+        private readonly string $user,
+        private readonly string $encPassword,
+    ) {
+        parent::__construct();
+        $host = rtrim($host, '/');
+        $this->host = $host === '' ? '' : $host . '/server/api/';
```

Comment:
does it make sense to have an empty host? Create a private function: host2ApiUrl that will throw improperaction if the host is empty string